composer/utils/module_surgery.py

```diff
@@ -178,6 +178,11 @@ def replace_module_classes(
             indices[policy_class] += 1
             if replacement is not None:
                 assert child not in replaced_pairs
+                # Preserve the device with surgery
+                # However, some children don't have any parameters, so using a for loop
+                for p in child.parameters():
+                    replacement = replacement.to(p.device)
+                    break
```

Comment:
```suggestion
                try:
                    p = next(child.parameters())
                    replacement = replacement.to(p.device)
                except:
                    pass
```

Reply:
Using `next` is more readable than `for p in.. break`, but is the try except needed?

Reply:
child.parameters() will always be a generator whether it has parameters or not. If you call `next` on a generator that is empty, you get a `StopIteration` exception, so if we want to go the `next` route we have to use the try except. We could also do:
```
if len(list(child.parameters())) > 0:
   p = next(child.parameters())
   replacement = replacement.to(p.device)
```
but that is more expensive and memory intensive because it actually loops through all the parameters and constructs a list.

Reply:
Sure, I could use `next` and just pass on `StopIteration` instead of using the for loop.

Reply:
Done.